### PR TITLE
backport electron.d.ts fixes 1.6

### PR DESCRIPTION
**Pull request by zeke**
This PR cherry picks some fixes to the API docs, the docs linter, and the typescript generator to bring 1.6's `electron.d.ts` file up to snuff.

🍒 

cc @kevinsawicki

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Should we include #9425 here as well?

**Comment by zeke** [comment]
> Should we include #9425 here as well?

Done, thanks. Hadn't gone far back enough in the history.

File: docs/api/browser-view.md [hunk 1]
```diff
@@ -46,29 +46,29 @@ Objects created with `new BrowserView` have the following properties:
 
 A [`webContents`](web-contents.md) object owned by this view.
```
Comment (kevinsawicki) [inline docs/api/browser-view.md]: I think this one was fixed in https://github.com/electron/electron/pull/9427 and should probably be backported as well.

Comment (zeke) [inline docs/api/browser-view.md]: ✅ 🍒

**Comment by kevinsawicki** [comment]
Thanks for backporting these @zeke 👍

**Pull request merged**